modules/external-api/src/main/java/org/opencastproject/external/endpoint/WorkflowDefinitionsEndpoint.java

```diff
@@ -0,0 +1,272 @@
+          @RestParameter(name = "withconfigurationpanel", description = "Whether the workflow configuration panel should be included in the response", isRequired = false, type = BOOLEAN),
+          @RestParameter(name = "filter", description = "A comma seperated list of filters to limit the results with. A filter is the filter's name followed by a colon \":\" and then the value to filter with so it is the form <Filter Name>:<Value to Filter With>.", isRequired = false, type = STRING),
+          @RestParameter(name = "sort", description = "Sort the results based upon a list of comma seperated sorting criteria. In the comma seperated list each type of sorting is specified as a pair such as: <Sort Name>:ASC or <Sort Name>:DESC. Adding the suffix ASC or DESC sets the order as ascending or descending order and is mandatory.", isRequired = false, type = STRING),
+          @RestParameter(name = "limit", description = "The maximum number of results to return for a single request.", isRequired = false, type = INTEGER),
+          @RestParameter(name = "offset", description = "The index of the first result to return.", isRequired = false, type = INTEGER) }, reponses = {
+          @RestResponse(description = "A (potentially empty) list of workflow definitions is returned.", responseCode = HttpServletResponse.SC_OK),
+          @RestResponse(description = "The request is invalid or inconsistent.", responseCode = HttpServletResponse.SC_BAD_REQUEST) })
+  public Response getWorkflowDefinitions(@HeaderParam("Accept") String acceptHeader,
+          @QueryParam("withoperations") boolean withOperations,
+          @QueryParam("withconfigurationpanel") boolean withConfigurationPanel, @QueryParam("filter") String filter,
+          @QueryParam("sort") String sort, @QueryParam("offset") Integer offset, @QueryParam("limit") Integer limit) {
+    Stream<WorkflowDefinition> workflowDefinitions;
+    try {
+      workflowDefinitions = workflowService.listAvailableWorkflowDefinitions().stream();
+    } catch (WorkflowDatabaseException e) {
+      logger.error("The workflow service was not able to get the workflow definitions: {}", getStackTrace(e));
+      return ApiResponses.serverError("Could not retrieve workflow definitions, reason: '%s'", getMessage(e));
+    }
+
+    // Apply filter
+    if (StringUtils.isNotBlank(filter)) {
+      for (String f : filter.split(",")) {
+        int sepIdx = f.indexOf(':');
+        if (sepIdx < 0 || sepIdx == f.length() - 1) {
+          logger.info("No value for filter {} in filters list: {}", f, filter);
+          continue;
+        }
+        String name = f.substring(0, sepIdx);
+        String value = f.substring(sepIdx + 1);
+
+        if ("tag".equals(name))
+          workflowDefinitions = workflowDefinitions.filter(wd -> ArrayUtils.contains(wd.getTags(), value));
+      }
+    }
+
+    // Apply sort
+    // TODO: this seems to not function as intended
```

Comment:
Why? Just did some basic tests and it seems to work as expected?

Reply:
I only did some quick manuell tests. Maybe I did something wrong. Glad it worked for you.

Reply:
@mtneug Note that I've already merged this PR...